Controller/AbstractFOSRestController.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Controller;
+
+use FOS\RestBundle\View\ViewHandlerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+/**
+ * Controllers using the View functionality of FOSRestBundle.
+ */
+abstract class AbstractFOSRestController extends AbstractController
+{
+    use ControllerTrait;
+
+    /**
+     * Get the ViewHandler.
+     *
+     * @return ViewHandlerInterface
+     */
+    protected function getViewHandler()
+    {
+        if (!$this->viewhandler instanceof ViewHandlerInterface) {
+            $this->viewhandler = $this->container->get('fos_rest.view_handler');
```

Comment:
We will also have to override the `getSubscribedServices()` method so that we can actually access the `fos_rest.view_handler` service here:

```php
public static function getSubscribedServices()
{
    return array(
            'fos_rest.view_handler' => ViewHandlerInterface::class,
    );
}
```

Reply:
Should we add `'fos_rest.view_handler' into parent list, like this?
```
public static function getSubscribedServices()
 {
     $subscribedServices = parent::getSubscribedServices();
     $subscribedServices['fos_rest.view_handler'] = ViewHandlerInterface::class;

     return $subscribedServices;
 }
```

Reply:
yes 👍